src/templates/staff/assignment.html

```diff
@@ -115,6 +132,30 @@
             }
         });
     }
+
+    function normalizeJSON(json) {
+        try {
+            return JSON.stringify(JSON.parse(json));
+        } catch {
+            return null;
+        }
+    }
+
+    function loadOriginalConfig() {
+        $.ajax({
+            type: "GET",
+            url: "{{ url_for('.staff_get_assignment_config', cid=course._id, aid=assignment.assignment_id) }}",
+            dataType: "json",
+            success: function (result) {
+                let json = JSON.stringify(result);
+                let old = normalizeJSON(json);
+                $('#mdl-edit-assn-config').val(old).data("old-config", old).removeAttr("disabled");
+            },
+            error: function () {
```

Comment:
If loading the old config failed, I think we should keep the config field disabled and show a message like "Config loading failed; you may proceed with editing other information". If something is broken, we should be in a fail-safe state and not allow config changes via on-demand as there is probably a version inconsistency between on-demand and API, or API is down.

Reply:
Yeah that makes sense